### PR TITLE
using collection summaries with cache at various places placed in the code

### PR DESCRIPTION
**Pull request by eishay**
@ymatsuda @andrewconner @raymondng @stkem someone please review
... loading from db the minimal we need

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/model/Collection.scala [hunk 1]
```diff
@@ -15,6 +15,7 @@ import play.api.libs.json._
 import com.keepit.serializer.BinaryFormat
 import java.io.{ByteArrayInputStream, DataInputStream, DataOutputStream, ByteArrayOutputStream}
 import scala.collection.mutable.ListBuffer
+import scala.Some
```
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/model/Collection.scala:18]: IntelliJ!

Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/model/Collection.scala:18]: lol

**Comment by andrewconner** [comment]
Big fan of this. Can the site's controller use this too?

**Comment by raymondng** [comment]
lgtm!

**Pull request merged**